File: SWEA/D2/1974. 스도쿠 검증/스도쿠 검증.py
T = int(input())
for t in range(1, T + 1):
    arr = [list(map(int, input().split())) for _ in range(9)]
    ans = 1
    
    for i in range(9):
        row = set()
        col = set()
        for j in range(9):
            row.add(arr[i][j])
            col.add(arr[j][i])
        if len(row) != 9 or len(col) != 9:
            ans = 0
            break
            
    if ans == 1:
        for i in range(0, 9, 3):
            for j in range(0, 9, 3):
                grid = set()
                for r in range(3):
                    for c in range(3):
                        grid.add(arr[i+r][j+c])
                if len(grid) != 9:
                    ans = 0
                    break
                    
    print(f"#{t} {ans}")

File: SWEA/D2/1974. 스도쿠 검증/스도쿠 검증.java
import java.util.Scanner;

public class Solution {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int t = 1; t <= T; t++) {
            int[][] arr = new int[9][9];
            for (int i = 0; i < 9; i++)
                for (int j = 0; j < 9; j++)
                    arr[i][j] = sc.nextInt();
                    
            int ans = 1;
            for (int i = 0; i < 9; i++) {
                int[] row = new int[10];
                int[] col = new int[10];
                for (int j = 0; j < 9; j++) {
                    if (++row[arr[i][j]] > 1) ans = 0;
                    if (++col[arr[j][i]] > 1) ans = 0;
                }
            }
            
            for (int i = 0; i < 9; i += 3) {
                for (int j = 0; j < 9; j += 3) {
                    int[] grid = new int[10];
                    for (int r = 0; r < 3; r++) {
                        for (int c = 0; c < 3; c++) {
                            if (++grid[arr[i+r][j+c]] > 1) ans = 0;
                        }
                    }
                }
            }
            System.out.println("#" + t + " " + ans);
        }
        sc.close();
    }
}